src/Microsoft.ML.TimeSeries/ExtensionsCatalog.cs

```diff
@@ -210,6 +210,55 @@ public static RootCause LocalizeRootCause(this AnomalyDetectionCatalog catalog,
             return dst;
         }
 
+        /// <summary>
+        /// <para>
+        /// In time series data, seasonality (or periodicity) is the presence of variations that occur at specific regular intervals,
+        /// such as weekly, monthly, or quarterly.
+        /// </para>
+        /// <para>
+        /// This method detects this predictable interval (or period) by adopting techniques of fourier analysis.
+        /// Assuming the input values have the same time interval (e.g., sensor data collected at every second ordered by timestamps),
+        /// this method takes a list of time-series data, and returns the regular period for the input seasonal data,
+        /// if a predictable fluctuation or pattern can be found that recurs or repeats over this period throughout the input values.
+        /// </para>
+        /// <para>
+        /// Returns -1 if no such pattern is found, that is, the input values do not follow a seasonal fluctuation.
+        /// </para>
+        /// </summary>
+        /// <param name="catalog">The detect seasonality catalog.</param>
+        /// <param name="input">Input DataView.The data is an instance of <see cref="Microsoft.ML.IDataView"/>.</param>
+        /// <param name="inputColumnName">Name of column to process. The column data must be <see cref="System.Double"/>.</param>
+        /// <param name="seasonalityWindowSize">An upper bound on the number of values to be considered in the input values.
+        /// When set to -1, use the whole input to fit model; when set to a positive integer, only the first windowSize number
+        /// of values will be considered. Default value is -1.</param>
+        /// <param name="randomnessThreshold"><a href ="https://en.wikipedia.org/wiki/Correlogram">Randomness flutuation threashold</a>
+        /// that specifies how confidence the input values follows a predictable pattern recurring over an interval as seasonal data.
+        /// By default, it is set as 2.81: <a href="https://en.wikipedia.org/wiki/Student%27s_t-distribution">two-sided t-distribution
```

Comment:
I an not sure I follow the meaning of this sentence. Can you please rephrase?

Reply:
rephrased and also takes your next comments: change randomness threshold to [0, 1] and default is 0.99 as confidence interval and internally maps it to [inverse normal cumulative distribution](http://www-personal.umich.edu/~andrewcb/DSO/Papers/Random%20Number%20Generator/notes_3.pdf) based on table, considering the randomness in auto-correlation.